pptxApp.py

```diff
@@ -96,67 +102,87 @@ def clean_text(text):
     return cleaned_text.strip()
 
 
-def save_explanations(explanations, presentation_path):
+def save_explanations(explanations, file_path):
     """
-    This method receives a list of strings each string implements an explanation, creates a JSON file and appends
-    the explanations to the file, in the format of (slide#: explanations).
-    :param: explanations: List of explanation retrieved from the API. (List of strings)
+    This method receives a list of strings, each string representing an explanation. It creates a JSON file and appends
+    the explanations to the file, using the original file name.
+    :param: explanations: List of explanations retrieved from the API. (List of strings)
+    :param: file_path: Path of the original file. (String)
     :return:
     """
-    # get the base name of power-point from path
-    presentation_name = os.path.basename(presentation_path)
-    # remove file's extension
-    presentation_name = os.path.splitext(presentation_name)[0]
-    # name the output file, with the name of power-point
-    output_file = f"{presentation_name}_explanations.json"
+    file_name = os.path.basename(file_path)
+    presentation_name, extension = os.path.splitext(file_name)
+    uid = re.search(r"\w{8}-\w{4}-\w{4}-\w{4}-\w{12}", presentation_name).group(0)
```

Comment:
If you count on a specific naming format then you'd need to check whether you actually receive compatible input.